File: year2/mpp/java/MotorcycleClientServer/MotorcycleCommon/src/main/java/validators/UserValidator.java
package validators;

import domain.User;

public class UserValidator implements Validator<User> {
    @Override
    public void validate(User user) throws ValidationException {
        if (user.getName() == null || user.getUsername() == null || user.getPassword() == null) {
            throw new ValidationException("All user fields must be not null");
        }
        if (user.getName().equals("") || user.getUsername().equals("") || user.getPassword().equals("")) {
            throw new ValidationException("All user fields must be not empty");
        }
    }
}

File: year2/mpp/c#/MotorcycleGrpcServer/MotorcycleCommon/Validators/UserValidator.cs
using MotorcycleContest.Domain.Entities;

namespace MotorcycleContest.Domain.Validators
{
    public class UserValidator : IValidator<User>
    {
        public void Validate(User user)
        {
            if (user.Name.Equals(null) && user.Username.Equals(null) || user.Password.Equals(null))
            {
                throw new ValidationException("null fields");
            }
            if (user.Name.Equals("") && user.Username.Equals("") || user.Password.Equals(""))
            {
                throw new ValidationException("empty fields");
            }
        }
    }
}
